src/java.base/share/classes/sun/launcher/LauncherHelper.java

```diff
@@ -861,6 +868,14 @@ private static Class<?> loadMainClass(int mode, String what) {
         } catch (LinkageError le) {
             abort(le, "java.launcher.cls.error4", cn,
                     le.getClass().getName() + ": " + le.getLocalizedMessage());
+        } finally {
+            if (jarFile != null) {
+                try {
+                    jarFile.close();
+                } catch (IOException ioe) {
+                    abort(ioe, "java.launcher.jar.error1", what);
```

Comment:
java.launcher.jar.error1 is "Error: An unexpected error occurred while trying to open file". I can't think of any cases where it might fail but the error message would be confusing if it did.

Reply:
Good catch! Introducing and using:
```properties
java.launcher.jar.error5=\
    Error: An unexpected error occurred while trying to close file {0}
```